database/seedGenerator.test.js

```diff
@@ -0,0 +1,55 @@
+const generateAllSeedData = require('./seedingUtils.js');
+
+describe('Seed Generation', () => {
+  describe('seed generation function', () => {
+    test('Creates one set of User and Project data', () => {
+      const single = generateAllSeedData(1);
+      expect(single.users.length).toBe(1);
+      expect(single.projects.length).toBe(1);
+    });
+
+    test('Creates multiple updates', () => {
+      const single = generateAllSeedData(1);
+      expect(single.updates.length).toBeGreaterThan(0);
+    });
+
+    test('Can create several sets of data', () => {
+      const SEED_SIZE = 10;
+      const several = generateAllSeedData(SEED_SIZE);
+      expect(several.users.length).toBe(SEED_SIZE);
+      expect(several.projects.length).toBe(SEED_SIZE);
+      expect(several.updates.length).toBeGreaterThan(SEED_SIZE);
+    });
+
+    test('Foreign key relationships are matched up correctly', () => {
+      const single = generateAllSeedData(1);
+      expect(single.users.userId).toBe(single.projects.ownerId);
+      expect(single.users.userId).toBe(single.updates.postedBy);
+      expect(single.projects.projectId).toBe(single.updates.projectId);
+    });
+
+    test('User data has the correct shape', () => {
+      const user = generateAllSeedData(1).users[0];
+      expect(user).toHaveProperty('userId', expect.any(Number));
+      expect(user).toHaveProperty('userName', expect.any(String));
+    });
```

Comment:
Maybe check for the presence/absence of additional properties. Shouldn't be too relevant in this context.

Reply:
That's a good idea, I'll see if I can figure out how to do that :)

Reply:
@KStockley figured out how to test for strict property matching (see below)